### PR TITLE
fix: fix dimension validation of lob_xml_host_variables

### PR DESCRIPTION
**Pull request by amanPras**
## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test analysis doesn't break for below code
```cobol
        Identification Division.
        Program-Id. 'TEST1'.
        Data Division.
         Working-Storage Section.
       01 GREET.
          02 VA USAGE IS SQL TYPE IS XML AS CLOB (10) OCCURS 400 TIMES.
        PROCEDURE DIVISION.
``` 

## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by amanPras** [comment]
@mm-broadcom please review

File: server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlHostVariable.java [hunk 1]
```diff
@@ -271,6 +271,15 @@ public class TestSqlHostVariable {
                   + "       04 {$*VAR`->VAR`->VAR-LENGTH`->VAR-DATA} USAGE IS SQL TYPE IS XML AS CLOB (10) OCCURS {123456|1} TIMES.\n"
                   + "        PROCEDURE DIVISION.\n";
 
+  public static final String LOB_XML_ARR_TEXT3_ERR2 =
+          "        Identification Division.\n"
+                  + "        Program-Id. 'TEST1'.\n"
+                  + "        Data Division.\n"
+                  + "         Working-Storage Section.\n"
+                  + "       01 {$*VAR-NAME1}.\n"
+                  + "       04 {$*VAR`->VAR`->VAR-LENGTH`->VAR-DATA} USAGE IS SQL TYPE IS XML AS CLOB-FILE OCCURS {123456|1} TIMES.\n"
```
Comment (slavek-kucera) [inline server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlHostVariable.java]: I can build the example without errors.
Also the transformation is a bit more complicated:
```
       04  VAR OCCURS 123456.                                           
         49 VAR-NAME-LENGTH PIC S9(9) COMP-5 SYNC.                      
         49 VAR-DATA-LENGTH PIC S9(9) COMP-5.                           
         49 VAR-FILE-OPTION PIC S9(9) COMP-5.                           
         49 VAR-NAME PIC X(255) SYNC.                                   
      *04 VAR USAGE IS SQL TYPE IS XML AS CLOB-FILE OCCURS 123456 TIMES.
```

Comment (amanPras) [reply server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlHostVariable.java]: updated.

File: server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlHostVariable.java [hunk 2]
```diff
@@ -279,11 +279,11 @@ public class TestSqlHostVariable {
           + "       01 {$*VAR-NAME1}.\n"
           + "       04 {$*VAR`->VAR`->VAR-NAME-LENGTH`->VAR-DATA-LENGTH`->VAR-FILE-OPTION`->VAR-NAME} USAGE IS SQL TYPE IS XML AS CLOB-FILE OCCURS {123456|1} TIMES.\n"
           + "        PROCEDURE DIVISION.\n"
-          + "           DISPLAY {$VAR}(1).\n"
-          + "           DISPLAY {$VAR-NAME-LENGTH}(1).\n"
-          + "           DISPLAY {$VAR-DATA-LENGTH}(1).\n"
-          + "           DISPLAY {$VAR-FILE-OPTION}(1).\n"
-          + "           DISPLAY {$VAR-NAME}(1).";
+          + "           DISPLAY {$VAR}.\n"
```
Comment (slavek-kucera) [inline server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlHostVariable.java]: The `(1)` should be there, the variables are arrays.

File: server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/sql/Db2SqlVisitor.java [hunk 3]
```diff
@@ -108,16 +108,57 @@ public List<Node> visitRowid_host_variables_arrays(Db2SqlParser.Rowid_host_varia
...
+        for (int i = 0; i < suffixes.length; i++) {
+            VariableNode variableNode = createVariableNode(variableDefinitionNode, generatedVariableLevel, suffixes[i], formats[i], usages[i]);
+            variableDefinitionNode.addChild(variableNode);
+            variableNode.addChild(new VariableDefinitionNameNode(variableDefinitionNode.getVariableName().getLocality(), variableNode.getName()));
+        }
+    }
+
+    private void addLobWithSizeNodes(VariableDefinitionNode variableDefinitionNode, int generatedVariableLevel) {
+        String[] suffixes = {"-LENGTH", "-DATA"};
+        String[] formats = {"S9(4)", "X(16)"};
```
Comment (slavek-kucera) [inline server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/sql/Db2SqlVisitor.java]:
```suggestion
        String[] formats = {"S9(4)", "X(" + len + ")"};
```

Comment (amanPras) [reply server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/sql/Db2SqlVisitor.java]: updated

File: server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlHostVariable.java (hunk 2 above)
Comment (amanPras) [reply server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/sql/TestSqlHostVariable.java]: updated

**Review by slavek-kucera — APPROVED** [review]

**Pull request merged**